app/inflation.go

```diff
@@ -0,0 +1,35 @@
+package app
+
+import (
+	"context"
+
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
+	minttypes "github.com/cosmos/cosmos-sdk/x/mint/types"
+	stakingkeeper "github.com/cosmos/cosmos-sdk/x/staking/keeper"
+	appparams "github.com/sourcenetwork/sourcehub/app/params"
+	tierkeeper "github.com/sourcenetwork/sourcehub/x/tier/keeper"
+)
+
+func ProvideInflationCalculationFn(
```

Comment:
question: Are we sure that this custom inflation function is actually necessary?

The tokenomics basically wants to compute the inflation based on total supply vs total locked (delegator stake + dev lock). 

![image](https://github.com/user-attachments/assets/6a06d390-4aac-4e6d-88d8-e54b7560b76f)

What we want to pass into the `minter.NextInflationFunction` at the end of this function is $R_n$ which is defined as $(stake_{delegator} + lock_{dev}) / total_{supply}$

My understanding of the custom function is that you've done is remove the $lock_{dev}$ from this calculation. Which would *increase* the inflation instead of what we want which is effectively *decreasing* inflation.

Reply:
I think the technically correctly updates the inflation ratio to be smaller as intended, but the problem is that the rewards aren't calculated correctly down stream, since the `tier` module account still gets minted rewards.

So we have correctly reduced the rewards, but now we distributed a smaller amount of rewards to the same number of delegated tokens.